### PR TITLE
New Installer WiX version 4

### PR DESCRIPTION
**Pull request by elaboy**
#### Using the FireGiant WiXV4 Visual Studio 2022 extension, right-click on the Boostrapper1 directory and select `Upgrade to WiX V4...`. Click `Okay` when the pop-up appears to convert the file format.

image

#### `DisplayInternalUI="yes"` has to be moved to the `bal` extension in WiXV4, as indicated by a wixtoolset collaborator. It used to be inside `MsiPackage`.
 https://github.com/wixtoolset/issues/issues/6164

image

## While Visual Studio still says `DisplayInternalUI attribute is not allowed`, the project builds without throwing any errors. I will open an issue in the wixtoolset repository explaining this situation. https://github.com/wixtoolset/issues/issues/6164

### CODE REVIEW

**Comment by acesnik** [comment]
* I tried building this, and it seems like it's still using `Wix 3` by default.
```
2>------ Build started: Project: Bootstrapper1, Configuration: Debug x86 ------
2>		C:\Program Files (x86)\WiX Toolset v3.11\bin\candle.exe -dDebug -d"DevEnvDir=C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\\" -dSolutionDir=C:\Users\User\source\repos\MetaMorpheus\ -dSolutionExt=.sln -dSolutionFileName=MetaMorpheus.sln -dSolutionName=MetaMorpheus -dSolutionPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheus.sln -dConfiguration=Debug -dOutDir=bin\Debug\ -dPlatform=x86 -dProjectDir=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\ -dProjectExt=.wixproj -dProjectFileName=Bootstrapper1.wixproj -dProjectName=Bootstrapper1 -dProjectPath=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\Bootstrapper1.wixproj -dTargetDir=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bin\Debug\ -dTargetExt=.exe -dTargetFileName=MetaMorpheusInstaller.exe -dTargetName=MetaMorpheusInstaller -dTargetPath=C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bin\Debug\MetaMorpheusInstaller.exe -dMetaMorpheusSetup.Configuration=Debug -d"MetaMorpheusSetup.FullConfiguration=Debug|x64" -dMetaMorpheusSetup.Platform=x64 -dMetaMorpheusSetup.ProjectDir=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\ -dMetaMorpheusSetup.ProjectExt=.wixproj -dMetaMorpheusSetup.ProjectFileName=MetaMorpheusSetup.wixproj -dMetaMorpheusSetup.ProjectName=MetaMorpheusSetup -dMetaMorpheusSetup.ProjectPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\MetaMorpheusSetup.wixproj -dMetaMorpheusSetup.TargetDir=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\bin\Debug\ -dMetaMorpheusSetup.TargetExt=.msi -dMetaMorpheusSetup.TargetFileName=MetaMorpheusInstaller.msi -dMetaMorpheusSetup.TargetName=MetaMorpheusInstaller -dMetaMorpheusSetup.TargetPath=C:\Users\User\source\repos\MetaMorpheus\MetaMorpheusSetup\bin\Debug\MetaMorpheusInstaller.msi -out obj\Debug\ -arch x86 -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixUtilExtension.dll" -ext "C:\Program Files (x86)\WiX Toolset v3.11\bin\\WixBalExtension.dll" bootstrapper1.wxs bootstrapper.wxproj
2>C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bootstrapper1.wxs(1,0): error CNDL0199: The Wix element has an incorrect namespace of 'http://wixtoolset.org/schemas/v4/wxs'.  Please make the Wix element look like the following: <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">.
2>C:\Users\User\source\repos\MetaMorpheus\Bootstrapper1\bootstrapper.wxproj(1,0): error CNDL0048: The document element name 'Project' is invalid.  A Windows Installer XML source file must use 'Wix' as the document element name.
2>Done building project "Bootstrapper1.wixproj" -- FAILED.
```
* Do you have any recommendation on how to get this to build by wix 4 instead of wix 3? It looks like there is some .wxproj that's maybe supposed to be added?
* Another request would be to finally change `Bootstrapper1` and `bootstrapper1` to just `Bootstrapper` and `bootstrapper`, since there's no other such thing here.

File: Bootstrapper1/bootstrapper1.wxs [hunk 1]
```diff
@@ -1,16 +1,12 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:bal="http://schemas.microsoft.com/wix/BalExtension" xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
```
Comment (trishorts) [inline Bootstrapper1/bootstrapper1.wxs:2]: Fix: Rename http://schemas.microsoft.com/wix/BalExtension to http://wixtoolset.org/schemas/v4/wxs/bal

**Comment by trishorts** [comment]
image

**Comment by trishorts** [comment]
image

Fix: Rename http://schemas.microsoft.com/wix/2006/wi to http://wixtoolset.org/schemas/v4/wxs

**Comment by trishorts** [comment]
image
'

File: CMD/CMD.csproj [hunk 2]
```diff
@@ -23,8 +23,8 @@
     <PackageReference Include="Microsoft.ML" Version="1.3.1" />
     <PackageReference Include="Microsoft.ML.FastTree" Version="1.3.1" />
     <PackageReference Include="Microsoft.NETCore.App" Version="2.2.8" />
-    <PackageReference Include="mzLib" Version="1.0.537" />
```
Comment (trishorts) [inline CMD/CMD.csproj:26]: mzlib and nett got accidentally shifted backwards. fix this.

File: GUI/TaskWindows/SearchTaskWindow.xaml [hunk 3]
```diff
@@ -495,7 +496,6 @@
                                 <CheckBox x:Name="WriteContaminantCheckBox" Content="Write contaminants" />
                                 <CheckBox x:Name="WriteIndividualResultsCheckBox" Content="Write individual file results" />
                                 <CheckBox x:Name="WriteSpectralLibraryCheckBox" Content="Write spectral library" />
-                                <CheckBox x:Name="UpdateSpectralLibraryCheckBox" Content="Update SpectralLibrary" />
```
Comment (trishorts) [inline GUI/TaskWindows/SearchTaskWindow.xaml:498]: this should not be missing

**Pull request closed without merging**